src/TextField/EnhancedTextarea.js

```diff
@@ -75,6 +75,12 @@ class EnhancedTextarea extends Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
We shouldn't need that hook. `componentWillReceiveProps` should be enough.
I think that the issue is coming from `syncHeightWithShadow` using the old maxHeight value and not the up coming one. Does that makes sens for you?